database/buildSchema/34B_review_assignable_functions.sql

```diff
@@ -26,8 +26,20 @@ CREATE OR REPLACE FUNCTION public.reviewable_questions (app_id int)
     WHERE
         ar.application_id = $1
         AND te.category = 'QUESTION'
-        AND (te.is_reviewable IS NULL
-            OR te.is_reviewable != 'NEVER')
+        AND ((ar.value IS NULL
+                AND te.is_reviewable = 'OPTIONAL_IF_NO_RESPONSE')
+            OR (ar.value IS NOT NULL
+                AND te.is_reviewable != 'NEVER'))
```

Comment:
The previous version didn't quite work to get all reviewable_questions considering what was:
- Submitted by applicant AND is_reviewable != 'NEVER'
- Not submitted by applicant AND is_reviewable = 'ALWAYS'